runtime/src/main/java/org/corfudb/runtime/view/AddressSpaceView.java

```diff
@@ -417,68 +421,51 @@ public void invalidateServerCaches() {
         });
     }
 
-    /** Force the client cache to be invalidated. */
+    /**
+     * Force the client cache to be invalidated.
+     */
     public void invalidateClientCache() {
         readCache.invalidateAll();
     }
 
     /**
      * Fetch a collection of addresses for insertion into the cache.
      *
-     * @param addresses collection of addresses to read from.
-     * @return A result to be cached
+     * @param addresses    collection of addresses to read from.
+     * @param waitForWrite flag whether wait for write is required or hole fill directly.
+     * @return a map of read addresses.
      */
-    public @Nonnull
-    Map<Long, ILogData> fetchAll(Iterable<Long> addresses) {
+    @Nonnull
+    public Map<Long, ILogData> fetchAll(Iterable<Long> addresses, boolean waitForWrite) {
         Map<Long, ILogData> result = new HashMap<>();
 
         Iterable<List<Long>> batches = Iterables.partition(addresses,
-            runtime.getParameters().getBulkReadSize());
+                runtime.getParameters().getBulkReadSize());
 
         for (List<Long> batch : batches) {
             try {
-                //doesn't handle the case where some address have a different replication mode
+                // Doesn't handle the case where some address have a different replication mode
                 result.putAll(layoutHelper(e -> e.getLayout()
```

Comment:
Should the API guarantee, that the set of addresses to be read is exactly the same set of addresses that were read?

What would happen if the client requests [1,2,3] and the server only returns [1, 3] ?

Reply:
Done. Added a sanity check.